src/transformers/models/speech_to_text/feature_extraction_speech_to_text.py

```diff
@@ -99,15 +99,12 @@ def utterance_cmvn(
         normalize_vars: Optional[bool] = True,
         padding_value: float = 0.0,
     ) -> np.ndarray:
-        # make sure we normalie float32 arrays
-        mean = x[:input_length].mean(axis=0)
-        square_sums = (x[:input_length] ** 2).sum(axis=0)
-
+        # make sure we normalize float32 arrays
         if normalize_means:
+            mean = x[:input_length].mean(axis=0)
             x = np.subtract(x, mean)
         if normalize_vars:
-            var = square_sums / x[:input_length].shape[0] - mean ** 2
-            std = np.sqrt(np.maximum(var, 1e-10))
+            std = x[:input_length].std(axis=0)
```

Comment:
Switched this logic to pure numpy to squeeze out a bit more precision when working with `np.float32` instead of casted `float->np.float64`